### ISSUE TITLE
Swagger org.springframework.core.io.Resource for 'file's 

### ISSUE THREAD

**Issue by LEgregius** [swagger-api/swagger-codegen#5089]
##### Description

The Spring framework support returning org.springframework.core.io.Resource interface implementations from rest api calls in order to process files in an abstract way, such as reading them from another url, database blob stream, etc, but the code gen only allows the type "file" to generate the class "java.io.File."   It would be nice to allow switch that to use the Resource interface.

##### Swagger-codegen version

Master

##### Swagger declaration file content or url

```yaml
/files/{fileId}:
    get:
      tags:
      description:
        Downloads an internal file
      parameters:
        - name: fileId
          in: path
          required: true
          type: string
      produces:
        - application/pdf
      responses:
        '200':
          description: The result
          schema:
            type: file
```

generates something like:

```java 
ResponseEntity<File> providerFilesFileIdGet(@ApiParam(value = "", required=true ) @PathVariable("fileId") String fileId);
```

But I need something like

```java
ResponseEntity<Resource> providerFilesFileIdGet(@ApiParam(value = "", required=true ) @PathVariable("fileId") String fileId);
```

Suggested Fix

I suggest adding an additional command line property to enable this feature.  I have already implemented one and will submit a PR for it.

**Comment by ePaul** [comment]
I would dare to say that using java.io.File (which is an object representing a file name in the local file system) is completely wrong here ... was this ever implemented, and how?
Would this create a local file with the returned data and return its name, or for a parameter, expect the user to create this file and pass it, then have the library read from the file?

I would have expected an InputStream or something similar.
(I don't know about your Resource either.)

**Comment by LEgregius** [comment]
I suppose if one were going to return a local file, as in return server file data, could make sense in some cases, but definitely not in any sort of large scale case.  The Resource is a SpringFramework thing.  It handles both opening the connection and returning the stream.

**Comment by wing328** [comment]
Closed via #5181
